compiler/luci/pass/src/SubstituteSqueezeToReshapePass.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteSqueezeToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/**
+ * @brief Return TRUE if all dim is known && dim > 0 && at least dim 1 exist
+ */
+bool can_squeeze_shape(luci::CircleNode *node)
+{
+  int dim_1_cnt = 0;
+  for (uint32_t r = 0; r < node->rank(); ++r)
+  {
+    if (not node->dim(r).known())
+      return false;
+    if (node->dim(r).value() < 1)
+      return false;
+    if (node->dim(r).value() == 1)
+      dim_1_cnt++;
```

Comment:
I think we should check the number of `dim==1` in the `squeeze_dims`.

Reply:
you mean `squeeze_dims` in `CircleSqueeze` ?

I'm not godd at these but what I understand is that
value of `squeeze_dims` is which dimension to squeeze.
so `dim == 1` check for that would doing "we want to squeeze dim 1"

Reply:
My comment was a bit confusing. What I meant was that we should check the number of dimensions equal to 1 among the dimensions in the `squeeze_dims`.

> here, we are checking if any input dim can be squeezed so, which would be having value 1

I think checking if any input dim can be squeezed is not enough. Let's see the below case.

input shape: [1, 2, 3, 4]
squeeze_dims: [1]
-> This case is not squeeze-able, because `squeeze_dims` in the input shape is 2 (!= 1). So, interpreter (or other runtime) will throw an exception when executing this operator. However, if we substitute this case, the network will run without an error. I thought this can be problematic.